modyn/supervisor/internal/triggers/trigger_datasets/online_trigger_dataset.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import random
+from typing import Generator, Optional
+
+from modyn.trainer_server.internal.dataset.online_dataset import OnlineDataset
+from torch.utils.data import IterableDataset
+
+logger = logging.getLogger(__name__)
+
+
+# TODO(#275): inherit common abstraction of dataset
+class OnlineTriggerDataset(OnlineDataset, IterableDataset):
```

Comment:
Could you add a docstring here explaining the idea of this dataset (sampling with some probability) and why we need it)?

Reply:
done.